### PR TITLE
fix: use sh instead of /bin/sh as shell fallback on Windows

### LINKED ISSUE #892: Workspace provisioning fails on Windows: spawn /bin/sh ENOENT
## Description

When using `git_worktree` execution workspace policy with a `provisionCommand`, the workspace runtime fails on Windows with:

```
spawn /bin/sh ENOENT
```

## Root Cause

In `server/src/services/workspace-runtime.ts`, the shell fallback is:

```typescript
const shell = process.env.SHELL?.trim() || "/bin/sh";
```

On Windows, `process.env.SHELL` is typically unset, so it falls back to `/bin/sh` which does not exist. This affects two call sites (workspace provision commands and runtime service spawning).

## Fix

Fall back to `"sh"` on Windows (available via Git Bash on PATH):

```typescript
const shell = process.env.SHELL?.trim() || (process.platform === "win32" ? "sh" : "/bin/sh");
```

## Environment

- OS: Windows 11
- Shell: Git Bash
- paperclipai: running from source (master)
- Workspace policy: `git_worktree` with `provisionCommand: "pnpm install"`

### PR DESCRIPTION
**Pull request by octasoft-ltd**
## Thinking Path

> - Paperclip orchestrates AI agents for zero-human companies
> - Agents run shell commands during workspace provisioning (git worktree creation, runtime services)
> - When `process.env.SHELL` is unset, the code falls back to `/bin/sh`
> - But on Windows with Git Bash, `/bin/sh` doesn't exist as an absolute path — Git Bash provides `sh` on PATH instead
> - This causes `child_process.spawn` to throw `ENOENT`, crashing workspace provisioning on Windows
> - This PR extracts a `resolveShell()` helper that uses `$SHELL` when set, falls back to `sh` (bare) on Windows or `/bin/sh` on Unix
> - The benefit is that agents running on Windows via Git Bash can provision workspaces without shell resolution errors
## Summary
- `workspace-runtime.ts` falls back to `/bin/sh` when `process.env.SHELL` is unset
- On Windows, `/bin/sh` doesn't exist → `spawn /bin/sh ENOENT`
- Fix: extract `resolveShell()` helper that uses `$SHELL` when set, falls back to `sh` on Windows (Git Bash PATH lookup) or `/bin/sh` on Unix

Three call sites updated to use the new helper.

Fixes #892

## Root cause

When Paperclip spawns shell commands in workspace operations (e.g., git worktree creation), it uses `process.env.SHELL` if set, otherwise defaults to `/bin/sh`. On Windows with Git Bash, `$SHELL` is typically unset and `/bin/sh` is not a valid path — Git Bash provides `sh` on PATH but not at the absolute `/bin/sh` location. This causes `child_process.spawn` to throw `ENOENT`.

## Approach

Rather than hard-coding a Windows-specific absolute path (e.g., `C:\Program Files\Git\bin\sh.exe`), we use the bare `"sh"` command which relies on PATH resolution. This works because:
1. Git Bash adds its `usr/bin` directory to PATH, making `sh` resolvable
2. On Unix/macOS, `/bin/sh` remains the correct default (it's the POSIX standard location)
3. `process.env.SHELL` takes priority when set, so this only affects the fallback

## Test plan

- [x] 7 unit tests for `resolveShell()`: SHELL set, trimmed, empty, whitespace-only, linux/darwin/win32 fallbacks
- [x] Run a workspace provision command on Windows with `git_worktree` strategy
- [x] Verify Unix/macOS is unaffected

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: server/src/services/workspace-runtime.ts [hunk 1]
```diff
@@ -273,7 +273,7 @@ async function runWorkspaceCommand(input: {
   env: NodeJS.ProcessEnv;
   label: string;
 }) {
-  const shell = process.env.SHELL?.trim() || "/bin/sh";
+  const shell = process.env.SHELL?.trim() || (process.platform === "win32" ? "sh" : "/bin/sh");
```
[the remark this answers is not in the document]
Comment (octasoft-ltd) [reply server/src/services/workspace-runtime.ts]: Addressed in PR #1115 which extracts a `resolveShell()` helper function (eliminating the duplication) and adds 7 unit tests for shell resolution edge cases.

**Comment by octasoft-ltd** [comment]
@greptileai review — added thinking path per CONTRIBUTING.md. Please re-score.

**Comment by devinfoley** [comment]
@greptileai
